astropy/wcs/src/wcslib_prjprm_wrap.c

```diff
@@ -0,0 +1,1025 @@
+        if (result) return result;
+        if (r0 != self->x->r0) {
+            self->x->r0 = r0;
+            self->x->flag = 0;
+            if (self->owner) ((PyCelprm*)self->owner)->x->flag = 0;
+        }
+    }
+    return 0;
+}
+
+
+static PyObject* PyPrjprm_get_phi0(PyPrjprm* self, void* closure)
+{
+    if (is_prj_null(self)) {
+        return NULL;
+    } else if (self->x->phi0 == UNDEFINED) {
+        Py_RETURN_NONE;
+    } else {
+        return get_double("phi0", self->x->phi0);
+    }
+}
+
+
+static int PyPrjprm_set_phi0(PyPrjprm* self, PyObject* value, void* closure)
+{
+    int result;
+    double phi0;
+    if (is_prj_null(self) || is_readonly(self)) {
+        return -1;
+    } else if (value == Py_None) {
+        if (self->x->phi0 != UNDEFINED) {
+            self->x->phi0 = UNDEFINED;
+            self->x->flag = 0;
+            if (self->owner) ((PyCelprm*)self->owner)->x->flag = 0;
+        }
+    } else {
+        result = set_double("phi0", value, &phi0);
+        if (result) return result;
+        if (phi0 != self->x->phi0) {
+            self->x->phi0 = phi0;
+            self->x->flag = 0;
+            if (self->owner) ((PyCelprm*)self->owner)->x->flag = 0;
+        }
+    }
+    return 0;
+}
+
+
+static PyObject* PyPrjprm_get_theta0(PyPrjprm* self, void* closure)
+{
+    if (is_prj_null(self)) {
+        return NULL;
+    } else if (self->x->theta0 == UNDEFINED) {
+        Py_RETURN_NONE;
+    } else {
+        return get_double("theta0", self->x->theta0);
+    }
+}
+
+
+static int PyPrjprm_set_theta0(PyPrjprm* self, PyObject* value, void* closure)
+{
+    int result;
+    double theta0;
+    if (is_prj_null(self) || is_readonly(self)) {
+        return -1;
+    } else if (value == Py_None) {
+        if (self->x->theta0 != UNDEFINED) {
+            self->x->theta0 = UNDEFINED;
+            self->x->flag = 0;
+            if (self->owner) ((PyCelprm*)self->owner)->x->flag = 0;
+        }
+    } else {
+        result = set_double("theta0", value, &theta0);
+        if (result) return result;
+        if (theta0 != self->x->theta0) {
+            self->x->theta0 = theta0;
+            self->x->flag = 0;
+            if (self->owner) ((PyCelprm*)self->owner)->x->flag = 0;
+        }
+    }
+    return 0;
+}
+
+
+static PyObject* PyPrjprm_get_pv(PyPrjprm* self, void* closure)
+{
+    int k;
+    Py_ssize_t size = PVN;
+    double *pv;
+    PyObject* pv_array;
+
+    if (is_prj_null(self)) {
+        return NULL;
+    }
+
+    pv = malloc(PVN * sizeof(double));
+    if (pv == NULL) {
+        PyErr_SetString(PyExc_MemoryError,
+                        "Could not allocate memory for PV array.");
+        return NULL;
+    }
+
+    for (k = 0; k < PVN; k++) {
+        if (self->x->pv[k] == UNDEFINED) {
+            pv[k] = (double) NPY_NAN;
+        } else {
+            pv[k] = self->x->pv[k];
+        }
+    }
+
+    pv_array = (PyArrayObject*) PyArray_SimpleNewFromData(1, &size, NPY_DOUBLE, pv);
```

Comment:
See comment below, but we may need to use PyArray_SetBaseObject here as well.

Reply:
same: new code will use `PyArray_SimpleNew()`